js/src/builtin/String.cpp

```diff
@@ -1898,7 +1908,7 @@ static bool str_normalize(JSContext* cx, unsigned argc, Value* vp) {
 
   // Latin-1 strings are already in Normalization Form C.
   if (form == NormalizationForm::NFC && str->hasLatin1Chars()) {
-    if (str->taint().hasTaint()) {
+    if (str->isTainted()) {
```

Comment:
Does this bring much performance? Perhaps we can also check whether these checks are/can be inlined.

Reply:
Oh, I noticed that the checks are non-uniform and thought I would unify them towards the simpler form where I noticed. I can revert those changes if that's preferable to you.

Inlining them might be a good idea, yeah.

Reply:
I think the ```str->isTainted()``` form is actually more elegant, so no need to revert the changes. I'll open an issue about inlining the tainting checks.